homeassistant/components/igloohome/lock.py

```diff
@@ -0,0 +1,91 @@
+"""Implementation of the lock platform."""
+
+from datetime import timedelta
+
+from aiohttp import ClientError
+from igloohome_api import (
+    BRIDGE_JOB_LOCK,
+    BRIDGE_JOB_UNLOCK,
+    DEVICE_TYPE_LOCK,
+    Api as IgloohomeApi,
+    ApiException,
+    GetDeviceInfoResponse,
+)
+
+from homeassistant.components.lock import LockEntity, LockEntityFeature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IgloohomeConfigEntry
+from .entity import IgloohomeBaseEntity
+from .utils import get_linked_bridge
+
+# Scan interval set to allow Lock entity update the bridge linked to it.
+SCAN_INTERVAL = timedelta(hours=1)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: IgloohomeConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up lock entities."""
+    async_add_entities(
+        IgloohomeLockEntity(
+            api_device_info=device,
+            api=entry.runtime_data.api,
+            bridge_id=str(
+                get_linked_bridge(device.deviceId, entry.runtime_data.devices)
+            ),
+        )
+        for device in entry.runtime_data.devices
+        if device.type == DEVICE_TYPE_LOCK
+        and get_linked_bridge(device.deviceId, entry.runtime_data.devices) is not None
+    )
```

Comment:
```suggestion
    async_add_entities(
        IgloohomeLockEntity(
            api_device_info=device,
            api=entry.runtime_data.api,
            bridge_id=str(bridge),
        )
        for device in entry.runtime_data.devices
        if device.type == DEVICE_TYPE_LOCK
        and (bridge := get_linked_bridge(device.deviceId, entry.runtime_data.devices)) is not None
    )
```
I think it should be possible to use a walrus here to avoid calling the method twice